src/validation.cpp

```diff
@@ -753,24 +770,9 @@ bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState& state, const C
 
     const Consensus::Params& consensus = Params().GetConsensus();
 
-    if (tx.IsZerocoinMint()) {
-        // Shows if old zerocoin mints are allowed yet in the mempool.
-        bool allow = (chainActive.Height() <= consensus.nSigmaStartBlock + consensus.nZerocoinV2MintMempoolGracefulPeriod);
-
-        if (!allow) {
-            return state.DoS(100, error("Old zerocoin mints no more allowed in mempool"),
-                             REJECT_INVALID, "bad-txns-zerocoin");
-        }
-    }
-
-    if (tx.IsZerocoinSpend()) {
-        // Shows if old zerocoin spends are allowed yet in the mempool.
-        bool allow = (chainActive.Height() <= consensus.nSigmaStartBlock + consensus.nZerocoinV2SpendMempoolGracefulPeriod);
-
-        if (!allow) {
-            return state.DoS(100, error("Old zerocoin spends no more allowed in mempool"),
-                             REJECT_INVALID, "bad-txns-zerocoin");
-        }
+    if (tx.IsZerocoinMint() || tx.IsZerocoinSpend() || tx.IsZerocoinRemint()) {
+        return state.DoS(100, error("Zerocoin transactions no more allowed in mempool"),
+                         REJECT_INVALID, "bad-txns-zerocoin");
     }
```

Comment:
if `CheckTransaction` rejects zerocoin transactions with `nHeight == INT_MAX` this check is not needed

Reply:
done.